airflow/models/taskinstance.py

```diff
@@ -1438,7 +1443,13 @@ def _execute_task_with_callbacks(self, context):
         """Prepare Task for Execution"""
         from airflow.models.renderedtifields import RenderedTaskInstanceFields
 
+        parent_pid = os.getpid()
+
         def signal_handler(signum, frame):
+            pid = os.getpid()
+            if pid != parent_pid:
+                os._exit(1)
```

Comment:
What's this for?

Reply:
So that's because ... well it's a bit of an odd-ball, but it's motivated by a DAG in the test suite which forks inside DAG code.

What happens then is that it returns back out to the calling framework, but we don't want to handle such a forked process coming back in, we just want to handle our own forked process.

Reply:
Comment explaining why needed then please.

Reply:
Added in 38d7acff3ccbbce2e2a5789f13221b336f705ac3.